arch/arm/core/swap_helper.S

```diff
@@ -182,6 +182,12 @@ _thread_irq_disabled:
     bic r3, #1
     orr r3, r0
     msr CONTROL, r3
+#ifndef CONFIG_CPU_CORTEX_M0
```

Comment:
If I have counted properly: this block is inside #ifdef CONFIG_ARMV7_M_ARMV8_M_MAINLINE, so we cannot have a Cortex-M0, i.e. the #ifdef guard is redundant.